pkg/resource/deploy/step_generator.go

```diff
@@ -592,6 +607,26 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, err
 		}
 	}
 
+	// If this is a destroy generation we're _always_ going to do a skip create or skip step here for
+	// custom non-provider resources.
```

Comment:
This could use a little bit of explanation *why* we're always going to do a skip create or skip step.  It's easy to see what this code is doing, but harder to understand why it is doing it.